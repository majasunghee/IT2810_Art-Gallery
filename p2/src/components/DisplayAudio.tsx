import React, {useState} from "react";
import audio1 from "../audio/Algorithms.mp3";
import audio2 from "../audio/Aspire.mp3"
import audio3 from "../audio/Slow_Vibing.mp3"
import audio4 from "../audio/The_Space_Inside.mp3"

function DisplayAudio(){
    const [audio, setAudio] = useState()

    const handleChange = (e: any) => {
       setAudio(e.target.value)
   }

return (

    <div>
        <select onChange={handleChange}>
            <option>Sounds</option>
            <option value={audio1}> 1 </option>
            <option value={audio2}> 2 </option>
            <option value={audio3}> 3 </option>
            <option value={audio4}> 4 </option>
        </select>
        <audio src={audio} autoPlay loop/>
    </div>
        )
}
export {DisplayAudio};

/*
import audio1 from "../audio/Algorithms.mp3"
import audio2 from "../audio/Aspire.mp3"
import audio3 from "../audio/Slow_Vibing.mp3"
import audio4 from "../audio/The_Space_Inside.mp3"
*/
