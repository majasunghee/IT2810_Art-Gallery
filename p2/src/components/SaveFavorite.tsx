import React from 'react';


function SaveFavorite() {

}

export {SaveFavorite};
