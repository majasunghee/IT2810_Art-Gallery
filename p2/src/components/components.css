body {
  background-color: black;
  align-items: center;
}

h1{
  border-bottom: solid 2px;
  border-color: white;
  color: white;
  padding-bottom: 30px;
  text-align: center;
}

.wrapper {
  margin: 10px;
  padding: 10px;
  display: grid;
  grid-template-columns: repeat(8, 1fr);
  grid-auto-rows: minmax(50px, auto);
}

.item_header {
  grid-column: 1/9;
}

.item_themeButton {
  grid-column: 1/2;
  text-align: end;
}

.item_audio {
  grid-column: 2/6;
  text-align: center;
}

.item_audioSelect {
  grid-column: 1/2;
  text-align: end;
}


.item_imgSelect {
  grid-column: 1/2;
  text-align: end;
}

.item_img {
  grid-column: 2/6;
}


.item_poem {
  grid-column: 6/8;
  text-align: center;
}

.content{
  text-align: center;
  margin-top: 50px;
  padding: 5px;
  border: 3px solid darkgrey;
  font-size: 14px;
}

.content > p{
  margin: 0;
}

.poem_title, .poem_author{
  font-weight: bold;
  font-variant: small-caps;
  padding-bottom: 10px;
}

.poem_title{
  font-size: 24px;
}
.poem_author{
  font-size: 14px;
  text-align: right;
}


#image_container{
  display: block;
  margin: 0px auto;
  width: 90%;
}

.drop_down, .drop_down > option {
  border: solid 2px;
  background-color: black;
  color: #F4F5F7;
  width: 6em;
  margin-top: 5px;
  font-size: 24px;
  font-family: courier;
  font-weight: bold;
  font-variant: small-caps;
  padding: 10px;
}

button{
  width: 6em;
  height: 2em;
  font-size: 24px;
  font-family: courier;
  font-weight: bold;
  font-variant: small-caps;
  padding: 5px;
  border: solid 2px #F4F5F7;
  text-align: center;
  color: #F4F5F7;
  background: black;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  font-weight: bold;
}

button:hover{
  border: solid 2px #F4F5F7;
  background: #F4F5F7;
  color: black;
}

/*Based on the size of an ipad pro*/
@media only screen and (max-width: 1024px) and (min-width: 750px) {

  h1 {
    font-size: 18px;
    padding-bottom: 20px;
  }

    .item_header {
    grid-column: 1/9;
  }

    .item_audio {
    grid-column: 3/7;
  }

    .item_img {
    grid-column: 3/7;
  }

    .item_poem {
    grid-column: 3/7;
  }

}

/*Based on the size of an iphone 6/7/8 Plus*/
@media only screen and (max-width: 750px) {

  .wrapper {
    margin: 5px;
    padding: 10px;
    display: grid;
    grid-template-columns: repeat(5, 1fr);
  }

  .item_audio {
    grid-column: 1/6;
    grid-row: 4/4;
  }

  .item_audioSelect {
    grid-column: 3/3;
  }


  .item_imgSelect {
    grid-column: 2/2;
    grid-row: 2/2;
  }

  .item_themeButton {
    grid-column: 1/2;
    text-align: end;
    margin-top: 1px;
  }

  .item_img {
    grid-column: 1/6;
  }

  .item_poem {
    grid-column: 2/5;
  }

  h1 {
      font-size: 18px;
      padding-bottom: 5px;
    }

  .drop_down, .drop_down > option {
    margin-top: 5px;
    font-size: 12px;
    padding: 2px;
  }

  button{
    width: 6em;
    padding: 2px;
    font-size: 12px;
    box-shadow: none;
  }

  .content{
    margin-top: 10px;
    padding: 5px;
    font-size: 12px;
  }

  .poem_title, .poem_author{
    padding-bottom: 5px;
  }

  .poem_title{
    font-size: 16px;
  }
  .poem_author{
    font-size: 10px;
  }

}