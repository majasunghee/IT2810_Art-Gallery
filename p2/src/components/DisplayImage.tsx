import React, {useContext} from 'react';
import './components.css'
import {useLocalStorage} from "../hooks/useLocalStorage"
import balloon from '../images/hotAirBalloonSVG.svg'
import seaplane from '../images/seaplaneSVG.svg'
import rocket from '../images/RocketSVG.svg'
import black from '../images/black.svg'
import ThemeContext from"../contexts/ThemeContext";

function DisplayImage(){

   const [image, setImage] = useLocalStorage('image', seaplane);

   const theme = useContext(ThemeContext);

   const handleChange = (e: any) => {
      setImage(e.target.value)
  }

  return(
     <>
         <div className="item_imgSelect" style={theme}>
           <select className="drop_down" onChange = {handleChange}>
           <option id="void_image" value={black} > images </option>
           <option value={balloon}>  balloon </option>
           <option value={seaplane}> seaplane </option>
           <option value={rocket}> rocket </option>
         </select>
         </div>
         <div className="item_img"><img id="image_container" src={image} alt="animated landscape"/></div>
     </>
  )

}

export {DisplayImage};

//<button onClick = {handleClick} > Get random Image </button>
//let randomNumb = Math.floor(Math.random()*3)
//getImage(imageArray[randomNumb])
