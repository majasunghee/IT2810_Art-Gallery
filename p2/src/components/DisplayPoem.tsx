import React, {useState, useEffect, useContext} from 'react';
import './components.css'
import {useLocalStorage} from "../hooks/useLocalStorage"
import ThemeContext from "../contexts/ThemeContext";

function DisplayPoem() {

  const theme = useContext(ThemeContext);

  const [listState, setListState] = useState()
  const [poem, getPoem] = useLocalStorage("poem", null)

  useEffect( () => {
    const result = fetch('https://poetrydb.org/linecount/10')
     .then(res => res.json())
     .then((data: any) =>{
        setListState(data);
     })
   }, []
   )

   const handleClick = () => {
       try {
           const randomNumb = Math.floor(Math.random() * listState.length)
           getPoem(listState[randomNumb]);
       }catch (err) {
           console.error(err);
       }
     }

  return (
    <div style={theme}>
    { !poem ?
      ( <div> loading... </div> ) :
      (
        <div className="content">
        <p className="poem_title"> {poem.title} </p>
          {poem.lines.map((item : any, index:any) => {
           return <p key={index}> {item } </p> })}
        <p className="poem_author"> {poem.author} </p>

        </div>
      )}
      <button onClick = {handleClick} > generate poem </button>
    </div>
)
}
export {DisplayPoem} ;
