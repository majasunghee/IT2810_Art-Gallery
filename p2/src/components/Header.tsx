import React, {useContext} from 'react';
import './components.css'
import ThemeContext from "../contexts/ThemeContext";

function Header() {
    const theme = useContext(ThemeContext);

    return (
        <div>
            <h1 style={theme}> Interaktiv Utstilling</h1>
        </div>
    )

}


export {Header}