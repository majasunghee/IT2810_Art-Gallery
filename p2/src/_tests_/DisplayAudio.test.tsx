import React from 'react';
import renderer from 'react-test-renderer';
import {DisplayAudio} from "../components/DisplayAudio";

it('renders correctly when there are no items', () => {
  const component = renderer.create(<DisplayAudio />).toJSON();
  expect(component).toMatchSnapshot();
});
