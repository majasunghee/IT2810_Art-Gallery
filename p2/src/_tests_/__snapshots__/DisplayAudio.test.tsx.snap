// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
Array [
  <div
    className="item_audioSelect"
    style={
      Object {
        "backgroundColor": "black",
        "borderColor": "white",
        "color": "white",
      }
    }
  >
    <select
      className="drop_down"
      onChange={[Function]}
    >
      <option>
        sounds
      </option>
      <option
        value="Algorithms.mp3"
      >
         space 
      </option>
      <option
        value="Aspire.mp3"
      >
         reminisce 
      </option>
      <option
        value="Slow_Vibing.mp3"
      >
         lofi 
      </option>
      <option
        value="The_Space_Inside.mp3"
      >
         lightsaber 
      </option>
      <option
        value="Videogame_Theme.mp3"
      >
         videogame 
      </option>
      <option
        value="Waterfall_And_Bird.mp3"
      >
         waterfall 
      </option>
    </select>
  </div>,
  <div
    className="item_audio"
  >
    <audio
      autoPlay={true}
      controls={true}
      loop={true}
      src={null}
    />
  </div>,
]
`;
